dsl_parser/functions.py

```diff
@@ -303,9 +292,10 @@ def validate(self, plan):
         args = [self.node_name] + self.property_path
         if any(is_function(x) for x in args):
             return
-        self.evaluate(plan)
+        handler = _PlanEvaluationHandler(plan, eager=True)
```

Comment:
what do u think about passing evaluate func handler and plan, so we would not re create _PlanEvaluationHandler in every call of validate func?

Reply:
there isn't an evaluation handler existing at this point yet, so we can't pass it in

it's very ugly that the validate method needs to evaluate, but this is what it did before, and i dont want to change it in this pr